### ISSUE TITLE
How do you maintain such a high quality of commit messages?

### ISSUE THREAD

**Issue by ewolfe** [kentcdodds/ama#343]
How do you maintain such a high quality of commit messages?

For example: 

```
docs(users): Add Mineral UI to list of Glamorous users (#357)
feat(forwardProps): allow to forward innerRef (#351)
feat(TypeScript): add filterProps to glamorous options (#340)
docs: add link to v4-announcement-blogpost (#341)
docs(users): add Rubrik to the glamorous users list (#338)
fix(umd): this fixes the UMD build for newer versions of React
chore(format): upgrade kcd-scripts and format all the things
fix(umd): get PropTypes from global object if using UMD build
chore(typescript): remove deprecated npm script (#336)
chore(build): fix the build to make sure we run things correctly
chore(build): update kcd-scripts
fix(umd): add sourcemaps to minified
fix(umd): add sourcemap
fix(propTypes): allow object as className (#333)
```

> https://github.com/paypal/glamorous/commits/master

Do you squash and merge each pull request and manually update each message title?

**Comment by newyork-anthonyng** [comment]
@ewolfe Tools like [commitizen](https://github.com/commitizen/cz-cli) might help too.

**Comment by kentcdodds** [comment]
Hi @ewolfe!

Like @newyork-anthonyng referenced, I actually used to use (and help maintain) commitizen which was helpful, but I found it unnecessary and just started writing my commit messages by hand.

For keeping the history clean with Pull Requests, I use GitHub's [Squash and Merge](https://help.github.com/articles/merging-a-pull-request/) feature so I can rewrite the commit message to match the format required by my automatic release process (with [semantic-release](https://github.com/semantic-release/semantic-release)).

I hope that's helpful!
